src/main/java/service/operation/ReturnOperation.java

```diff
@@ -0,0 +1,23 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class ReturnOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public ReturnOperation() {
+        this.fruitDao = new FruitDaoImpl();
+    }
+
+    @Override
+    public void handle(String fruit, int quantity) {
+        int balanceAfterReturn = fruitDao.getFruitBalance(fruit) + quantity;
+        if (balanceAfterReturn >= 0) {
```

Comment:
The check for a non-negative balance is not necessary in the `ReturnOperation` class because returning items should only increase the balance. You may safely remove this condition.